regionServiceClient/lib/cloudregister/registerutils.py

```diff
@@ -0,0 +1,419 @@
+# Copyright (c) 2015, SUSE LLC, All rights reserved.
+#
+# This library is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 3.0 of the License, or (at your option) any later version.
+# This library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# Lesser General Public License for more details.
+# You should have received a copy of the GNU Lesser General Public
+# License along with this library.
+
+"""Utility functions for the cloud guest registration"""
+
+import base64
+import glob
+import logging
+import os
+import pickle
+import requests
+import stat
+import subprocess
+import sys
+
+from cloudregister import smt
+from M2Crypto import X509
+
+REGISTRATION_DATA_DIR = '/var/lib/cloudregister/'
+REGISTERED_SMT_SERVER_DATA_FILE_NAME = 'currentSMTInfo.obj'
+AVAILABLE_SMT_SERVER_DATA_FILE_NAME = 'availableSMTInfo_%d.obj'
+
+
+# ----------------------------------------------------------------------------
+def check_registration(smt_server_name):
+    """Check if the instance is already registerd"""
+    if has_repos(smt_server_name) and __has_credentials(smt_server_name):
+        return 1
+
+    return None
+
+
+# ----------------------------------------------------------------------------
+def clean_hosts_file(domain_name):
+    """Remove the smt server entry from the /etc/hosts file"""
+    new_hosts_content = []
+    content = open('/etc/hosts', 'r').readlines()
+    smt_announce_found = None
+    for entry in content:
+        if '# Added by SMT' in entry:
+            smt_announce_found = True
+            continue
+        if smt_announce_found and domain_name in entry:
+            smt_announce_found = False
+            continue
+        new_hosts_content.append(entry)
+
+    hosts_file = open('/etc/hosts', 'w')
+    for entry in new_hosts_content:
+        hosts_file.write(entry)
+    hosts_file.close()
```

Comment:
This is purely style, so I'm not going to get upset if you leave it, but I prefer

```
with open(HOSTSFILE, 'w') as host_file:
    for entry in new_hosts_content:
        hosts_file.write(entry)
```